flake: cli Test_Tasks/delete_task

Spotted while running a standard CI job:

```
=== FAIL: cli Test_Tasks/delete_task (0.44s)
    clitest.go:75: invoking command: coder --global-config /var/folders/lf/wbx6llps6z37wkfpk85_942m0000gn/T/Test_Tasksdelete_task1504307573/001 exp task delete funny-nobel233 --yes
    exp_task_test.go:143: 
        	Error Trace:	/Users/runner/work/coder/coder/cli/exp_task_test.go:143
        	Error:      	Received unexpected error:
        	            	running command "coder exp task delete": delete task "naughty-gagarin8-zJ3/funny-nobel233":
        	            	    github.com/coder/coder/v2/cli.(*RootCmd).taskDelete.func1
        	            	        /Users/runner/work/coder/coder/cli/exp_task_delete.go:75
        	            	  - DELETE http://127.0.0.1:50142/api/experimental/tasks/naughty-gagarin8-zJ3/e3e442e4-03f4-4287-9814-4cc170c9034c: unexpected status code 500: Failed to delete task
        	            		Error: sql: no rows in result set
        	Test:       	Test_Tasks/delete_task
    --- FAIL: Test_Tasks/delete_task (0.44s)
```

--- a/cli/exp_task_test.go
+++ b/cli/exp_task_test.go
@@ -53,18 +53,6 @@ func Test_Tasks(t *testing.T) {
 		taskName     = strings.ReplaceAll(testutil.GetRandomName(t), "_", "-")
 	)
 
-	// The below test cases are not real "sub-tests" as they depend on each other sequentially.
-	runTest := func(name string, cmdArgs []string, assertFn func(string, *codersdk.Client)) {
-		var stdout strings.Builder
-		inv, root := clitest.New(t, cmdArgs...)
-		inv.Stdout = &stdout
-		clitest.SetupConfig(t, userClient, root)
-		require.NoError(t, inv.WithContext(ctx).Run(), name)
-		if assertFn != nil {
-			assertFn(stdout.String(), userClient)
-		}
-	}
-
 	for _, tc := range []struct {
 		name     string
 		cmdArgs  []string
@@ -146,7 +134,15 @@ func Test_Tasks(t *testing.T) {
 			},
 		},
 	} {
-		runTest(tc.name, tc.cmdArgs, tc.assertFn)
+		t.Logf("test case: %q", tc.name)
+		var stdout strings.Builder
+		inv, root := clitest.New(t, tc.cmdArgs...)
+		inv.Stdout = &stdout
+		clitest.SetupConfig(t, userClient, root)
+		require.NoError(t, inv.WithContext(ctx).Run(), tc.name)
+		if tc.assertFn != nil {
+			tc.assertFn(stdout.String(), userClient)
+		}
 	}
 }
 

--- a/cli/exp_task_test.go
+++ b/cli/exp_task_test.go
@@ -53,7 +53,18 @@ func Test_Tasks(t *testing.T) {
 		taskName     = strings.ReplaceAll(testutil.GetRandomName(t), "_", "-")
 	)
 
-	//nolint:paralleltest // The sub-tests of this test must be run sequentially.
+	// The below test cases are not real "sub-tests" as they depend on each other sequentially.
+	runTest := func(name string, cmdArgs []string, assertFn func(string, *codersdk.Client)) {
+		var stdout strings.Builder
+		inv, root := clitest.New(t, cmdArgs...)
+		inv.Stdout = &stdout
+		clitest.SetupConfig(t, userClient, root)
+		require.NoError(t, inv.WithContext(ctx).Run(), name)
+		if assertFn != nil {
+			assertFn(stdout.String(), userClient)
+		}
+	}
+
 	for _, tc := range []struct {
 		name     string
 		cmdArgs  []string
@@ -135,16 +146,7 @@ func Test_Tasks(t *testing.T) {
 			},
 		},
 	} {
-		t.Run(tc.name, func(t *testing.T) {
-			var stdout strings.Builder
-			inv, root := clitest.New(t, tc.cmdArgs...)
-			inv.Stdout = &stdout
-			clitest.SetupConfig(t, userClient, root)
-			require.NoError(t, inv.WithContext(ctx).Run())
-			if tc.assertFn != nil {
-				tc.assertFn(stdout.String(), userClient)
-			}
-		})
+		runTest(tc.name, tc.cmdArgs, tc.assertFn)
 	}
 }
 

